java/vector/src/main/codegen/templates/BaseWriter.java

```diff
@@ -61,20 +61,25 @@ public interface StructWriter extends BaseWriter {
 
```

Comment:
This is where some design decisions have to be made of how to actually expose the `ExtensionWriter` to the user and how the user should actually extend some `ExtensionWriter`. 

As mentioned above I would make this one interface `BaswWriter` exposing all the possible writers and leaving them abstract in the concrete implementations.

Reply:
This sounds reasonable to me.